napalm_ios/ios.py

```diff
@@ -1134,129 +1136,225 @@ def bgp_time_conversion(bgp_uptime):
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Description: (?P<description>.+)'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+BGP version \d+, remote router ID '
+                                      r'(?P<remote_id>\d+(\.\d+){3})'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+For address family: (?P<afi>\S+) (?P<safi>\S+)'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Prefixes Current:\s+(?P<sent_prefixes>\d+)\s+'
+                                      r'(?P<accepted_prefixes>\d+).*'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Saved (soft-reconfig):.+(?P<received_prefixes>\d+).*'),
+                 'record': True},
+                {'regexp': re.compile(r'^\s+Local Policy Denied Prefixes:.+'),
+                 'record': True}
+            ],
+            'no_fill_fields': ['received_prefixes']
+        }
+        # parse outputs into a list of dicts
+        summary_data = []
+        summary_data_entry = {}
+        for line in summary_output.splitlines():
+            for item in parse_summary['patterns']:
+                match = item['regexp'].match(line)
+                if match:
+                    summary_data_entry.update(match.groupdict())
+                    if item['record']:
+                        summary_data.append(copy.deepcopy(summary_data_entry))
+                        for field in parse_summary['no_fill_fields']:
+                            try:
+                                del summary_data_entry[field]
+                            except KeyError:
+                                pass
+                    break
+        neighbor_data = []
+        neighbor_data_entry = {}
+        for line in neighbor_output.splitlines():
+            for item in parse_neighbors['patterns']:
+                match = item['regexp'].match(line)
+                if match:
+                    neighbor_data_entry.update(match.groupdict())
+                    if item['record']:
+                        neighbor_data.append(copy.deepcopy(neighbor_data_entry))
+                        for field in parse_neighbors['no_fill_fields']:
+                            try:
+                                del neighbor_data_entry[field]
+                            except KeyError:
+                                pass
+                    break
+        router_id = None
+        # check that we got a list of dicts
+        assert summary_data and isinstance(summary_data, list)
+        for entry in summary_data:
+            assert isinstance(entry, dict)
+            assert 'router_id' in entry
+            # select the first router_id value
+            if not router_id:
+                router_id = entry['router_id']
+            # check that every other value matches
+            assert entry['router_id'] == router_id
```

Comment:
Probably just covert this to an else statement and raise a value error. If line 1244 happened, then this will be true (so else is probably more logical flow).